### PR TITLE
Support more peripherals

### PR DESCRIPTION
**Pull request by mateusz-holenko**
This PR adds support for new LiteX peripherals:

* ethernet PHY
* cpu timer
* control and status

### CODE REVIEW

File: generate-renode-scripts.py [hunk 1]
```diff
@@ -231,7 +231,7 @@ def generate_spiflash(peripheral, shadow_base, **kwargs):
         string: repl definition of the peripheral
     """
 
-    flash_size = 0x2000000
+    flash_size = 0x10000000
```
Comment (mithro) [inline generate-renode-scripts.py]: Flash size should come from the platform?

File: generate-renode-scripts.py [hunk 2]
```diff
@@ -151,9 +163,16 @@ def generate_cpu():
     kind = constants['config_cpu_type']['value']
 
     if kind == 'VEXRISCV':
```
Comment (mithro) [inline generate-renode-scripts.py:169]: Privilege support is only correct on the Linux configuration of vexriscv.

**Review by mithro — APPROVED** [review]
Generally looks good to me.

Couple of minor comments.

File: generate-renode-scripts.py (hunk 1 above)
Comment (mateusz-holenko) [reply generate-renode-scripts.py]: I noticed that not only `flash_size`, but also xip base address were hardcoded in the script. I fixed both by using values from the `csv` file.

File: generate-renode-scripts.py (hunk 2 above)
Comment (mateusz-holenko) [reply generate-renode-scripts.py:169]: Now privilege architecture and `A` instruction set support is enabled only for cpu `LINUX` variant.

File: generate-renode-scripts.py [hunk 3]
```diff
@@ -142,18 +154,36 @@ def generate_silencer(peripheral, shadow_base, **kwargs):
...
 cpu: CPU.VexRiscv @ sysbus
 """
+        if variant == 'LINUX':
+            result += """
+    cpuType: "rv32ima"
+    privilegeArchitecture: PrivilegeArchitecture.Priv1_10
+"""
+        else:
+            result += """
+    cpuType: "rv32im"
```
Comment (mithro) [inline generate-renode-scripts.py:180]: I'm not sure when `m` is enabled?

**Review by mithro — APPROVED** [review]
LGTM!

File: generate-renode-scripts.py (hunk 3 above)
Comment (mateusz-holenko) [reply generate-renode-scripts.py:180]: According to [this code](https://github.com/enjoy-digital/litex/blob/master/litex/soc/cores/cpu/vexriscv/core.py#L35), it is enabled for variant "standard" and up.

Comment (mithro) [reply generate-renode-scripts.py:180]: So, should probably match that...

I wonder if that information should end up in the headers at all?

Comment (mateusz-holenko) [reply generate-renode-scripts.py:180]: You can obtain this information by reading `MISA` CSR in RISC-V, so from the software-perspective it does not have to be included in the headers, I think.

Comment (mithro) [reply generate-renode-scripts.py:180]: @mateusz-holenko But that doesn't help compile time optimization?

**Pull request merged**